resources/todomvc/architecture-examples/lit/src/lib/todos.ts

```diff
@@ -0,0 +1,160 @@
+export interface Todo {
+	id: number;
+	text: string;
+	completed: boolean;
+}
+
+const todoFilters = ["all", "active", "completed"] as const;
+export type TodoFilter = (typeof todoFilters)[number];
+function isTodoFilter(value: string | undefined): value is TodoFilter {
+	return todoFilters.includes(value as TodoFilter);
+}
+
+/**
+ * A mutable, observable container for a todo list.
+ *
+ * @fires a `change` event when the todo list changes.
+ */
+export class Todos extends EventTarget {
+	#nextId;
+	#todos: Array<Todo>;
+	#filter: TodoFilter = this.#filterFromUrl();
+	constructor() {
+		super();
+		const stored = window.localStorage.getItem("todos");
```

Comment:
I'm worried about the use of `localStorage`.

1. Is this use typical in real-world applications? This involves a lot of JSON.parse/stringify as well.
2. This may be a problem with the repeatibility of the benchmark, we need to make sure that the data in the storage is empty at the start of each run.

What do other folks think? @rniwa @bgrins @camillobruni @flashdesignory

Reply:
some of the original todoMVC apps (todomvc.com) use local storage. 
We don't use that in the speedometer specific todo apps though.

Reply:
There have been concerns raised in the past about including `localStorage` or other APIs that hit disc in terms of I/O dominating measurements - so even if they are widely used APIs the current guidance is to not use them. I think we could revisit that decision if there was a strong functional need for them in a workload, but that's not the case here so I would drop it.